manifests/run.pp

```diff
@@ -385,18 +387,24 @@
 
   if $restart {
     if $ensure == 'absent' {
+      $restart_stop_command = [$docker_command, 'stop', "--time=${stop_wait_time}", $sanitised_title]
+      $restart_stop_onlyif = [$docker_command, 'inspect', $sanitised_title]
+
       exec { "stop ${title} with docker":
-        command     => "${docker_command} stop --time=${stop_wait_time} ${sanitised_title}",
-        onlyif      => "${docker_command} inspect ${sanitised_title}",
+        command     => $restart_stop_command,
+        onlyif      => $restart_stop_onlyif,
         environment => $exec_environment,
         path        => $exec_path,
         provider    => $exec_provider,
         timeout     => $exec_timeout,
       }
 
+      $restart_remove_command = "${docker_command} rm -v ${sanitised_title}"
+      $restart_remove_onlyif = [$docker_command, 'inspect', $sanitised_title]
```

Comment:
I don't think this will work as expected. onlyif will be expecting an array of arrays.

```suggestion
      $restart_remove_onlyif = [[$docker_command, 'inspect', $sanitised_title]]
```

Reply:
Same as above. Will try to implement.